workspace/php2/src/XSolveSecurityBundle/Controller/UserController.php

```diff
@@ -19,17 +20,70 @@ public function RestLoginAction(Request $request) {
       $autorizationManager = $this->get('x_solve_security.xsolve_autorization_manager');
       $authenticatedToken = $autorizationManager->getToken($username, $password);
 
-      if ($authenticatedToken) {
-	 $this->getUser()->assignNewToken();
-	 $this->getDoctrine()->getManager()->flush();
+      if (!$authenticatedToken) {
+
+	 throw new AccessDeniedException('unable to find token');
+      }
+      $this->getUser()->assignNewToken();
+      $this->getDoctrine()->getManager()->flush();
+
+      return (new JsonResponse())->setData(
+		      ['token' => $this->getUser()->getToken()->getToken()]);
+   }
+   public function showUserAction($id) {
+
+      $user = $this->getDoctrine()
+	      ->getRepository('XSolveSecurityBundle:User')
+	      ->find($id);
+
+      if (!$user) {
+
+	 throw $this->createNotFoundException(
+		 'No found user');
+      }
+
+      return (new JsonResponse())->setData(
+		      ['login' => $user->getUsername(),
+			'password' => $user->getPassword()]);
+   }
+   public function showAllUsersAction() {
+
+      $users = $this->getDoctrine()
+	      ->getRepository('XSolveSecurityBundle:User')
+	      ->findAll();
+
+      $tab = []; //tab contain id, usernames and passwords of all user in database
 
+      foreach ($users as &$user) {
 
-	 return (new JsonResponse())->setData(
-			 ['login' => $username,
-			  'password' => $username,
-		          'token' => $this->getUser()->getToken()->getToken()]);
+	 $element = ['id' => $user->getId(),
+	     'username' => $user->getUsername(), ];
+	 array_push($tab, $element);
       }
-      throw new AccessDeniedException('unable to find token');
+
+
+      $response = new jsonResponse();
+      $response->setData($tab);
+
+      return $response;
+   }
+
+   public function registerAction(Request $request) {
+      
+      $username = $request->request->get('login');
+      $password = $request->request->get('haslo');
+      $user = new User();
+      $user->setUsername($username);
+      $user->setPassword($password);
+
+      $em = $this->getDoctrine()->getManager(); //add user
+      $em->persist($user);	  //to database
+      $em->flush();
+
+      return (new JsonResponse())->setData(
+		      ['login' => $user->getUsername(),

```

Comment:
Technicznie, mógłbyś dodać tutaj model który będzie przyjmował obiekt ``User`` i posiadał metodę getViewData() który zwróci właśnie taką tablicę. Pozwoli to wyciągnąć odpowiedzialność za przygotowywanie tej tablicy do jednego wspólnego miejsca co w przyszłości zaprocentuje, jednym miejsce zmiany dla wszystkich metod które oczekują tych danych.

```
$userModel = new UserModel($user);
$response->setData($userModel->getViewData());

return $response;